### PR TITLE
Modular

### PR DESCRIPTION
**Pull request by mramitdas**
# Pull Request

**Description**

User model -> Refactor User class to inherit from Base
- Updated the User class to inherit from the newly created Base class, which provides generic database operations.
- Removed duplicated code related to database interactions within the User class.
- The User class now leverages the shared functionality provided by the Base class.
- This change enhances code maintainability and reduces redundancy.

Subscription model -> Refactor Subscription class to inherit from Base
- Updated the Subscription class to inherit from the newly created Base class, which provides generic database operations.
- Removed duplicated code related to database interactions within the Subscription class.
- The Subscription class now leverages the shared functionality provided by the Base class.
- This change enhances code maintainability and reduces redundancy.

Please provide a brief description of the changes made in this pull request.

**Related Issue(s)**

- Closes #<issue_number>
- Fixes #<issue_number>

**Type of Change**

- [✅] Bug fix
- [✅] New feature
- [✅] Enhancement
- [✅] Documentation update
- [ ] Other (please describe)

**Checklist**

Please check off the following items before submitting your pull request:

- [✅] Code is properly formatted (You can use tools like Black for Python)
- [ ] Tests have been added or updated to cover the changes
- [✅] Documentation has been updated to reflect the changes
- [✅] Code follows project coding guidelines

**Additional Notes**

Provide any additional information or context that may be helpful for reviewing the pull request.

### CODE REVIEW

File: app/schemas/utils.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 from datetime import datetime
+
```
Comment (techonerd) [inline app/schemas/utils.py:2]: unwanted change.

File: app/api/V1/endpoints/login.py
Comment (techonerd) [inline app/api/V1/endpoints/login.py:1]: why a blank file? removed empty file?

**Review by techonerd — APPROVED** [review]
looks good. some unwanted changes are present but can be taken care in following changes.

File: app/schemas/utils.py (hunk 1 above)
Comment (mramitdas) [reply app/schemas/utils.py:2]: pep8 convention followed.

File: app/api/V1/endpoints/login.py
Comment (mramitdas) [reply app/api/V1/endpoints/login.py:1]: it was initiated earlier but later reaslised login functionality can be written in user file itself.

**Pull request merged**